src/Core/src/ImageSources/UriImageSourceService/UriImageSourceService.iOS.cs

```diff
@@ -21,20 +23,35 @@ public partial class UriImageSourceService
 
 			try
 			{
-				// TODO: use a real caching library with the URI
-				if (imageSource is not IStreamImageSource streamImageSource)
-					return null;
+				var hash = Crc64.ComputeHashString(imageSource.Uri.OriginalString);
+				var pathToImageCache = CacheDirectory + hash + ".png";
+
+				NSData? imageData;
+
+				if (IsImageCached(pathToImageCache))
```

Comment:
We should check here to see if caching is enabled for the image source... Android shows how to check the setting: https://github.com/dotnet/maui/blob/12f17cfc85e7ee8e7ad3a215d4ab69befb18fa5c/src/Core/src/ImageSources/UriImageSourceService/UriImageSourceService.Android.cs#L32

Probably faster to check this than to look if the file exists, if caching is disabled anyway.